docs/programming.rst

```diff
@@ -66,35 +86,63 @@ terminal:
     $ python -m psij plugins
 
 
-The executor implementations provided by this version of PSI/J Python
+JobExecutor Base Class
+^^^^^^^^^^^^^^^^^^^^^^
+
+The ``psij.JobExecutor`` class is abstract, but offers concrete static methods
```

Comment:
Static methods are always concrete.

Reply:
If you mean that as a general statement about all static methods that's not true, https://stackoverflow.com/a/31590500/10563162

Reply:
Ha! I forget we live in pythonworld.

Reply:
> > I think it is meant to go with a removal of the sphinx auto-generated structure, since that would duplicate the class details, but in the default way that sphinx does it.
> 
> No, not necessarily. We can have both, and this PR leaves in the auto-generated structure. At the bottom of the API reference page is a link to the complete autogenerated documentation.

True, that is also an option. I didn't consider it because it duplicates class documentation in multiple places, which I didn't think it's something we want to do. But clearly that's an option that we can discuss.

If we want to be exhaustive, there is also a fourth option :)

Reply:
why?!?